controller/resource_test.go

```diff
@@ -345,7 +345,6 @@ func TestTaskStateReconcile(t *testing.T) {
 					Partition: exp.Partition,
 				})
 				require.Error(t, err)
-				require.Contains(t, err.Error(), "403 (ACL not found)")
```

Comment:
This error string is only applicable to Consul versions prior to 1.15 so I decided to just take it out.

Reply:
minor: I think all both pre-1.15 and post-1.15 versions of Consul contain `ACL not found`, just in a different parts of the error message? It would be good to still check we are getting the correct type of error.

Reply:
Great catch Paul. You're correct that both versions have `ACL not found`. I've updated the test to re-instate the check.